### PR TITLE
web3-eth-accounts: Update to latest version of uuid package

### PR DESCRIPTION
**Pull request by LucianBuzzo**
This change updates to the latest version of uuid, to fix an issue with Math.random() (displayed in a warning on install).
Older versions may use Math.random() in certain circumstances, which is known to be problematic.
See https://v8.dev/blog/math-random for details.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:unit` with success.
- [x] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

**Comment by LucianBuzzo** [comment]
Because the `web3-eth-accounts` module uses `uuid` with the `random` option, it appears to be unaffected by the issue described in my commit message. However it doesn't hurt to update the dependency and prevent a potentially misleading warning message from appearing.

**Comment by jdevcs** [comment]
@LucianBuzzo Thanks for pointing this. 

Please could you remove all changes of your PR in  

- dist/web3.min.js 
- dist/web3.min.js.map and 
- packages/web3-eth-accounts/package-lock.json .

**Review by jdevcs — CHANGES_REQUESTED** [review]
@LucianBuzzo Thanks for pointing this.

Please could you remove all changes of your PR in

- dist/web3.min.js
- dist/web3.min.js.map and
- packages/web3-eth-accounts/package-lock.json .

File: packages/web3-eth-accounts/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
         "eth-lib": "0.2.8",
         "ethereumjs-util": "^7.0.10",
         "scrypt-js": "^3.0.1",
-        "uuid": "3.3.2",
+        "uuid": "8.3.2",
```
Comment (nazarhussain) [inline packages/web3-eth-accounts/package.json:23]: That's a drastic change. Let's hope nothing breaks 🤞

Comment (LucianBuzzo) [reply packages/web3-eth-accounts/package.json:23]: @nazarhussain Are there specific things you are concerned about that we could add test cases for? What do you think the biggest risk factors are?

**Comment by LucianBuzzo** [comment]
@jdevcs Done, btw, should the dist files be generated in an `npm prepack` stage instead of committed to version control?

**Comment by jdevcs** [comment]
> @jdevcs Done, btw, should the dist files be generated in an `npm prepack` stage instead of committed to version control?

Thanks @LucianBuzzo , We dnt commit dist files in feature, bug fix or lib-update PRs. 

@spacesailor24 @luu-alex Review/approval for uuid lib update.

**Comment by luu-alex** [comment]
Hey thank you for this PR :) from what it seems in the codebase we only use `uuid` [once](https://github.com/ChainSafe/web3.js/blob/2279a67e0702343764db5cae2dffc04048083952/packages/web3-eth-accounts/src/index.js#L598). It does look like the method is affected by the version upgrade but it is not breaking. 
Perhaps adding a unit testcase using `uuid.v4()` with the attribute random as an arg wouldn't hurt. Thoughts?

Also updating the [changelog](https://github.com/ChainSafe/web3.js/blob/1.x/CHANGELOG.md)  and detailing the change

**Comment by jdevcs** [comment]
@LucianBuzzo please merge down 1.x branch into your branch and fix merge conflicts. Thanks

**Review by avkos — APPROVED** [review]

**Review by jdevcs — COMMENTED** [review]
This change is addressed in https://github.com/web3/web3.js/pull/5529 so I am closing this PR.

**Pull request closed without merging**